File: flex-fields/src/Dignite.Abp.FlexFields.Abstractions/Dignite/Abp/FlexFields/FlexFieldValueType.cs
namespace Dignite.Abp.FlexFields;

/// <summary>
/// Which typed slot a field's value belongs in: it is how a provider types the values
/// <see cref="IFieldType.GetSearchableValues"/> hands back, and it describes a
/// <see cref="FlexFieldQueryCondition.Value"/> so the query side can compare against that same slot.
/// </summary>
public enum FlexFieldValueType
{
    String,
    Number,
    DateTime,
    Boolean,
    Guid
}

File: flex-fields/angular/projects/flex-fields/src/lib/models/flex-field-value-type.ts
/**
 * Which typed slot a field's value belongs in. Mirrors the `FlexFieldValueType` enum on the server —
 * the ordinals are the wire format, so the order here is fixed.
 */
export enum FlexFieldValueType {
  String = 0,
  Number = 1,
  DateTime = 2,
  Boolean = 3,
  Guid = 4,
}
